File: documentation-examples-kotlin/src/kotlin/dev/typr/foundations/docs/sqlserver/DomainType.kt
package dev.typr.foundations.docs.sqlserver

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*

@Suppress("unused")
class DomainType {
    //start
    // Wrapper type
    data class OrderId(val value: Int)

    // Create SqlServerType from INT
    val orderIdType: SqlServerType<OrderId> =
        SqlServerTypes.int_.bimap(::OrderId, OrderId::value)
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/sqlserver/DomainType.scala
package dev.typr.foundations.docs.sqlserver
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*



@SuppressWarnings(Array("unused"))
object DomainType:
  //start
  // Wrapper type
  case class OrderId(value: Int)

  // Create SqlServerType from INT
  val orderIdType: SqlServerType[OrderId] =
    SqlServerTypes.int_.bimap(OrderId.apply, _.value)
  //stop
